lib/chef/provider/apt_update.rb

```diff
@@ -75,6 +77,13 @@ def do_update
         declare_resource(:execute, "apt-get -q update")
       end
 
+      def self.uses_apt?
+        ENV["PATH"] ||= ""
+        paths = %w{ /bin /usr/bin /sbin /usr/sbin } + ENV["PATH"].split(::File::PATH_SEPARATOR)
+        paths.any? { |path| ::File.executable?(::File.join(path, "apt-get")) }
```

Comment:
there's another task here to clean up the code duplication in uses_apt? between the two providers, and use the which helper and/or fix the which helper, along with ideally adding uses_apt? to some kind of DSL helper so that we can use it everywhere, but that's getting well outside of the scope of what i wanted to do here and raises namespace concerns and i'm just not feeling creative enough today to handling naming things...